browser/ui/views/view_shadow.h

```diff
@@ -0,0 +1,79 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_UI_VIEWS_VIEW_SHADOW_H_
+#define BRAVE_BROWSER_UI_VIEWS_VIEW_SHADOW_H_
+
+#include "base/memory/raw_ptr.h"
+#include "base/memory/raw_ref.h"
+#include "ui/compositor/layer.h"
+#include "ui/compositor/layer_delegate.h"
+#include "ui/gfx/shadow_value.h"
+#include "ui/views/view_observer.h"
+
+// Manages a layer that will render a drop shadow under a given view. When an
+// instance is constructed for a view, the view will be set to paint to a layer
+// (if not already set) and a shadow layer will be added below. To remove the
+// shadow, release the `ViewShadow` instance.
+//
+// Example:
+//
+//   class ViewWithShadow : public views::View {
+//    private:
+//     constexpr int kCornerRadius = 8;
+//
+//     constexpr ViewShadow::ShadowParameters kShadow{
+//         .offset_x = 0,
+//         .offset_y = 1,
+//         .blur_radius = 4,
+//         .shadow_color = SkColorSetA(SK_ColorBLACK, 0.07 * 255)};
+//
+//     ViewShadow shadow_{this, kCornerRadius, kShadow};
+//   };
+//
+class ViewShadow : public views::ViewObserver, public ui::LayerDelegate {
```

Comment:
Should ViewShadow extend ui::LayerOwner too, as it owns the ui::Layer?

Reply:
Good question. I'd prefer not to provide the full `LayerOwner` interface to users (since, for now, users don't need to recreate the layer or anything like that). But it does make sense to use an instance of `LayerOwner` to manage the layer internally.